### PR TITLE
Add infoboxes for Slayer bracelets.

### PR DESCRIPTION
**Pull request by Reasel**
This adds an option to display infoboxes for slayer bracelets.

There is a toggle for this option that is defaulted to off.

villainouswarpedelk-max-14mb

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java [hunk 1]
```diff
@@ -304,6 +316,33 @@ private void checkInventories()
 		slayerItems = ImmutableList.copyOf(items);
 	}
 
+	@Subscribe
+	public void onItemContainerChanged(final ItemContainerChanged event)
+	{
+		if (event.getItemContainer() == client.getItemContainer(InventoryID.EQUIPMENT))
+		{
+			final Stream<Item> items = Arrays.stream(event.getItemContainer().getItems());
+			hasExpeditiousEquip = (items.anyMatch(item -> itemManager.getItemComposition(item.getId()).getId() == ItemID.EXPEDITIOUS_BRACELET));
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java]: We know the index of the bracelet... just check that?

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java [hunk 2]
```diff
@@ -304,6 +316,33 @@ private void checkInventories()
...
+			final Stream<Item> seconds = Arrays.stream(event.getItemContainer().getItems());
+			hasSlaughterEquip = (seconds.anyMatch(item -> itemManager.getItemComposition(item.getId()).getId() == ItemID.BRACELET_OF_SLAUGHTER));
+			save();
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java]: why?

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java [hunk 3]
```diff
@@ -449,14 +499,27 @@ private void killedOne()
...
+			addCounter();
+			counter.setText(String.valueOf(amount));
+		}
+
+		if (config.showBraceletInfoBox())
+		{
+			if (hasExpeditiousEquip || hasExpeditiousInvent)
+			{
+				addExpeditiousCounter();
+				expeditiousCounter.setText(String.valueOf(expeditiousChargeCount));
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java]: The charges are not changed here... instead it should update the infobox when it does change

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java [hunk 4]
```diff
@@ -466,7 +529,11 @@ private void setTask(String name, int amt)
 		amount = amt;
 		save();
 		removeCounter();
+		removeSlaughterCounter();
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java]: Setting task has no effect on bracelet charges?

**Comment by Reasel** [comment]
I refactored a lot of the code once I realized what you mean Adam. Things should be a lot slimmer and maker more sense visually.

**Comment by Reasel** [comment]
Closing this PR and will open a new proper PR after the deob removal.

**Pull request closed without merging**